app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java

```diff
@@ -149,6 +158,60 @@ public void afterTextChanged(Editable s) {}
     return view;
   }
 
+  private String getZimNoteDirectoryName() {
+    String zimFileName = ZimContentProvider.getZimFile(); // Returns name of the form ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim"
+
+    String noteDirectoryName = getTextAfterLastSlashWithoutExtension(zimFileName);
+
+    if(noteDirectoryName != null) {
+      return noteDirectoryName;
+    } else {
+      return (zimFileTitle); // Incase the required ZIM file name couldn't be extracted
+    }
+  }
+
+  private String getArticleNotefileName() {
+    String articleUrl = ((MainActivity) getActivity()).getCurrentWebView().getUrl(); // Returns url of the form: "content://org.kiwix.kiwixmobile.zim.base/A/Main_Page.html"
+
+    String notefileName = getTextAfterLastSlashWithoutExtension(articleUrl);
+
+    if(notefileName != null) {
+      return notefileName;
+    } else {
+      return (articleTitle); // Incase the required html file name couldn't be extracted
+    }
+  }
+
+  private String getTextAfterLastSlashWithoutExtension(String path) {
+    /* That's about exactly what it does.
+     *
+     * From ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim", returns "granbluefantasy_en_all_all_nopic_2018-10"
+     * From "content://org.kiwix.kiwixmobile.zim.base/A/Main_Page.html", returns "Main_Page"
+     * For null input or on being unable to find required text, returns null
+     * */
+    if(path == null) return null;
+
+    int rightmostSlash = -1;
+    int rightmostDot = -1;
+
+    for(int i = path.length()-1; i >= 0; i--) {
```

Comment:
this can be replaced by using String.lastIndexOf

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1237/commits/085ed7f7bd8c0916f1d1a3ad25a705a3bf1bdfd4#diff-36e13fa4a9132616aee81e1f88bfa0ffR185)